### DISCUSSION TITLE
How advanced is the Text Replacer regex?

### DISCUSSION THREAD

**Discussion by tobiasvl** [ObsidianPublisher/obsidian-github-publisher#51] [Q&A — Plugin] [answered]
Hey! Just asking here, since it's a bit cumbersome to test this in detail by making changes to both files and settings, then pushing and merging, etc.

How advanced is the regex support in the Text Replacer? Specifically...

* Does it work with multiple lines, or just single lines?
* Can it reference capture groups?

I would love to, for example, replace callouts with actual HTML (divs with a specific class), since Jekyll Kramdown doesn't support callouts. But that seems a bit beyond its capabilities. The problem is that I don't know what its capabilities are.

Thanks!

**Comment by Mara-Li** [comment]
Normally, capture group can work (another user use it !)
For multiple line, i need to check the code or test. Normally, it must works on it, but i don't test this !

**Reply by tobiasvl** [reply]
I looked at [the code](https://github.com/ObsidianPublisher/obsidian-github-publisher/blob/faf9007fb4c27dd9fe5735aca2b49da729621fec/plugin/contents_conversion/findAndReplaceText.ts#L19). It seems [the multiline flag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/multiline) isn't enabled, which would make working with multiple lines easier, although [it should still be possible](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#using_regular_expression_on_multiple_lines).

I'm not sure if that's something you'd be open to, but I opened a PR: #52 

Perhaps it should be documented which flags are enabled?

**Comment by Mara-Li** [comment, marked as answer]
Hello!
Last beta version (-6) add a way to set custom tags for regex!
![image](https://user-images.githubusercontent.com/30244939/204373511-92f1ab1a-86bd-4938-a7ca-39cfd7e495ec.png)
By default, the flags are fixed to `gi` (global + case insensitive)
